Alicerce/Sources/Persistence/DiskMemoryPersistence.swift

```diff
@@ -0,0 +1,326 @@
+            return false
+        }
+
+        return true
+    }
+
+    private func directoryContents(with keys: [URLResourceKey]) -> [URL] {
+        guard let url = URL(string: configuration.path) else {
+            assertionFailure("💥 failed to create url with path 👉 `\(configuration.path)`")
+            return []
+        }
+
+        guard let urls = try? fileManager.contentsOfDirectory(at: url,
+                                                              includingPropertiesForKeys: keys,
+                                                              options: .skipsPackageDescendants)
+            else {
+                assertionFailure("💥 failed to read directory content for path 👉 \(url.absoluteString)")
+                return []
+        }
+
+        return urls
+    }
+
+    // MARK: - Operations
+
+    private func createEvictOperation() -> DiskMemoryBlockOperation {
+        return DiskMemoryBlockOperation(qos: .background) { [unowned self] in
+
+            // Check if should run eviction
+            guard self.configuration.diskLimit < self.usedDiskSize else {
+                return
+            }
+
+            let extraOccupiedDiskSize = self.usedDiskSize - self.configuration.diskLimit
+
+            let urls = self.directoryContents(with: [.contentAccessDateKey, .fileSizeKey])
+
+            typealias FileAccessTimeSizeTuple = (accessTime: TimeInterval, size: UInt64)
+            typealias FilePathAttributesTuple = (path: String, fileAttr: FileAccessTimeSizeTuple)
+
+            let accessTimes: [FilePathAttributesTuple] = urls.map {
+                let resourceValue = $0.resourceValues(forKeys: [.contentAccessDateKey, .fileSizeKey])
+
+                return ($0.path, (resourceValue.contentAccessDate?.timeIntervalSince1970 ?? 0, UInt64(resourceValue.fileSize ?? 0)))
+                }.sorted { $0.fileAttr.accessTime > $1.fileAttr.accessTime }
+
+            var evictSize: UInt64 = 0
+
+            let filesToRemove = accessTimes.filter {
+                guard evictSize < extraOccupiedDiskSize else {
+                    return false
+                }
+
+                evictSize += $0.fileAttr.size
+
+                return true
+            }
+
+
+            filesToRemove.map {$0.path}
+            .forEach {
+                guard let _ = try? self.fileManager.removeItem(atPath: $0) else {
+                    assertionFailure("💥 Failed to remove file with path 👉 \($0)")
+
+                    return print("💥 Failed to remove file with path 👉 \($0)")
+                }
+            }
+        }
+    }
+}
+
+fileprivate final class DiskMemoryBlockOperation: BlockOperation {
+    
+    required init(qos: QualityOfService? = .default, block: @escaping () -> Swift.Void) {
+        super.init()
+        
+        addExecutionBlock(block)
+        if let qos = qos { qualityOfService = qos }
+    }
+}
+
+fileprivate extension URL {
```

Comment:
Perhaps this extension is useful for other scenarios? If so we could move it to an :sparkles:`URL.swift`:sparkles: extension 😄